src/util/modules.py

```diff
@@ -666,13 +744,23 @@ def _caller_version(depth=2):
 
 
 def force_upgrade_of_fresh_module(cr, module, init=True):
-    """It may appear that new (or forced installed) modules need an upgrade script to grab data
-    from another module. (we cannot add a pre-init hook on the fly)
+    """
+    Force the execution of upgrade scripts for a module that is being installed.
 
-    Being in init mode may make sens in some situations (when?) but has the nasty side effect
-    of not respecting noupdate flags (in xml file nor in ir_model_data) which can be quite
-    problematic
+    Standard Odoo doesn't run upgrade scripts when a module is being installed. This makes
```

Comment:
Fun fact: the first versions actually did it. But it caused so much trouble that I removed this: https://github.com/odoo/odoo/commit/4ec71c74d1dcec96d764fb42aab856cb32fb5321